src/update-feeds-loops/submit-transactions.ts

```diff
@@ -196,12 +298,25 @@ export const submitTransactions = async (
   api3ServerV1: Api3ServerV1,
   updatableDataFeeds: UpdatableDataFeed[],
   blockNumber: number
-) =>
-  Promise.all(
-    updatableDataFeeds.map(async (dataFeed) =>
-      submitTransaction(chainId, providerName, provider, api3ServerV1, dataFeed, blockNumber)
-    )
-  );
+) => {
+  const {
+    config: { walletDerivationScheme },
+  } = getState();
+  if (walletDerivationScheme.type === 'fixed') {
+    if (isEmpty(updatableDataFeeds)) {
```

Comment:
Imo, this should be handled in `submitBatchTransaction`. Also, make it a single line return.

Reply:
I'm now thinking if it's even worth calling [submitTransactions](https://github.com/api3dao/airseeker-v2/blob/single-sponsor-wallet/src/update-feeds-loops/update-feeds-loops.ts#L294) when this array is empty 🤔

Reply:
Doesn't matter to me.